keyboards/darkproject/kd87a_bfg_edition/keymaps/via/keymap.c

```diff
@@ -0,0 +1,186 @@
+  KC_GESC,  KC_BRID,  KC_BRIU,  KC_MC,    KC_SIRI,  KC_F5,    KC_F6,    KC_MRWD,  KC_MPLY,  KC_MFFD,  KC_MUTE,    KC_VOLD,  KC_VOLU,  KC_PSCR,  KC_SCRL, KC_PAUS,   
+  KC_GRV,   KC_1,     KC_2,     KC_3,     KC_4,     KC_5,     KC_6,     KC_7,     KC_8,     KC_9,     KC_0,       KC_MINS,  KC_EQL,   KC_BSPC,  KC_INS, KC_HOME, KC_PGUP,  
+  KC_TAB,   KC_Q,     KC_W,     KC_E,     KC_R,     KC_T,     KC_Y,     KC_U,     KC_I,     KC_O,     KC_P,       KC_LBRC,  KC_RBRC,  KC_BSLS,  KC_DEL, KC_END, KC_PGDN,    
+  KC_CAPS,  KC_A,     KC_S,     KC_D,     KC_F,     KC_G,     KC_H,     KC_J,     KC_K,     KC_L,     KC_SCLN,    KC_QUOT,            KC_ENT,       
+  KC_LSFT,  KC_Z,     KC_X,     KC_C,     KC_V,     KC_B,     KC_N,     KC_M,     KC_COMM,  KC_DOT,   KC_SLSH,    KC_RSFT,            KC_UP,      
+  KC_LCTL,  KC_LALT,  KC_LGUI,                      KC_SPC,                       KC_RALT,  MO(Macfn),  KC_APP,   KC_RCTL,  KC_LEFT,  KC_DOWN,  KC_RGHT),
+
+[Winfn] = LAYOUT(
+  KC_GESC,  KC_MYCM,  KC_MAIL,  KC_WSCH,  KC_WHOM,  KC_CALC,  KC_MSEL,  KC_MPRV,  KC_MPLY,  KC_MNXT,  KC_MUTE,    KC_VOLD,  KC_VOLU,  KC_PSCR,  KC_SCRL, KC_PAUS,  
+  KC_GRV,   KC_1,     KC_2,     KC_3,     KC_4,     KC_5,     KC_6,     KC_7,     KC_8,     KC_9,     KC_0,       KC_MINS,  KC_EQL,   KC_BSPC,  KC_INS,  RGB_HUI, KC_PGUP,  
+  KC_TAB,   KC_Q,     TO(Win),  KC_E,     KC_R,     KC_T,     KC_Y,     KC_U,     KC_I,     KC_O,     KC_P,       KC_LBRC,  KC_RBRC,  KC_BSLS,  RGB_MOD, KC_END, KC_PGDN,   
+  KC_CAPS,  KC_A,     KC_S,     KC_D,     KC_F,     KC_G,     KC_H,     KC_J,     KC_K,     KC_L,     KC_SCLN,    KC_QUOT,            KC_ENT,      
+  KC_LSFT,  KC_Z,     KC_X,     KC_C,     KC_V,     KC_B,     KC_N,     TO(Mac),  KC_COMM,  KC_DOT,   KC_SLSH,    KC_RSFT,            RGB_VAI,    
+  KC_LCTL,  GUI_TOG,  KC_LALT,                      KC_SPC,                       KC_RALT,  MO(Winfn),  KC_APP,   KC_RCTL,  RGB_SPD,  RGB_VAD,  RGB_SPI),
+
+[Macfn] = LAYOUT(
+  KC_GESC,  KC_F1,    KC_F2,    KC_F3,    KC_F4,    KC_F5,    KC_F6,    KC_F7,    KC_F8,    KC_F9,    KC_F10,     KC_F11,   KC_F12,   KC_PSCR,  KC_SCRL, KC_PAUS, 
+  KC_GRV,   KC_1,     KC_2,     KC_3,     KC_4,     KC_5,     KC_6,     KC_7,     KC_8,     KC_9,     KC_0,       KC_MINS,  KC_EQL,   KC_BSPC,  KC_INS,  RGB_HUI, KC_PGUP,  
+  KC_TAB,   KC_Q,     TO(Win),  KC_E,     KC_R,     KC_T,     KC_Y,     KC_U,     KC_I,     KC_O,     KC_P,       KC_LBRC,  KC_RBRC,  KC_BSLS,  RGB_MOD, KC_END, KC_PGDN,    
+  KC_CAPS,  KC_A,     KC_S,     KC_D,     KC_F,     KC_G,     KC_H,     KC_J,     KC_K,     KC_L,     KC_SCLN,    KC_QUOT,            KC_ENT,      
+  KC_LSFT,  KC_Z,     KC_X,     KC_C,     KC_V,     KC_B,     KC_N,     TO(Mac),  KC_COMM,  KC_DOT,   KC_SLSH,    KC_RSFT,            RGB_VAI,     
+  KC_LCTL,  KC_LALT,  KC_LGUI,                      KC_SPC,                       KC_RALT,  MO(Winfn),  KC_APP,   KC_RCTL,  RGB_SPD,  RGB_VAD,  RGB_SPI),
+
+};
+
+bool rgb_matrix_indicators_user(void) {
```

Comment:
This isn't the right function for this. 

You can use the `LED_CAPS_LOCK_PIN` define in config.h to handle this. 

As for the layers, `layer_state_set_user` would be the better place for that.